atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/KvsSweepQueueTest.java

```diff
@@ -195,6 +188,169 @@ public void sweepDeletesTombstonesWhenWriteHasHigherTimestamp() {
+        long finePartitionForSweepTs = tsPartitionFine(provider.getSweepTimestamp(Sweeper.CONSERVATIVE));
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+
+        assertProgressUpdatedToFineTimestampPartition(finePartitionForSweepTs - 1L);
+    }
+
+    @Test
+    public void sweepProgressesToPartitionOfJustSwept() {
+        long finePartitionForSweepTs = tsPartitionFine(provider.getSweepTimestamp(Sweeper.CONSERVATIVE));
+        long writeTs = provider.getSweepTimestamp(Sweeper.CONSERVATIVE) - 3 * TS_FINE_GRANULARITY;
+        enqueueWrite(TABLE_CONSERVATIVE, writeTs);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertProgressUpdatedToFineTimestampPartition(tsPartitionFine(writeTs));
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertProgressUpdatedToFineTimestampPartition(finePartitionForSweepTs - 1L);
+
+        assertThat(tsPartitionFine(writeTs)).isLessThan(finePartitionForSweepTs);
+    }
+
+    @Test
+    public void sweepCellOnlyOnceWhenInLastPartitionBeforeSweepTs() {
+        immutableTs = 2 * TS_COARSE_GRANULARITY - TS_FINE_GRANULARITY;
+        verify(kvs, never()).deleteAllTimestamps(any(TableReference.class), anyMap());
+
+        enqueueWrite(TABLE_CONSERVATIVE, immutableTs - 1);
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        verify(kvs, times(1)).deleteAllTimestamps(any(TableReference.class), anyMap());
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        verify(kvs, times(1)).deleteAllTimestamps(any(TableReference.class), anyMap());
+    }
+
+    @Test
+    public void sweepableTimestampsGetsScrubbedWhenNoMoreToSweepButSweepTsInNewCoarsePartition() {
+        long tsSecondPartitionFine = TS + TS_FINE_GRANULARITY;
+        long largestFirstPartitionCoarse = TS_COARSE_GRANULARITY - 1L;
+        enqueueWrite(TABLE_CONSERVATIVE, TS);
+        enqueueWrite(TABLE_CONSERVATIVE, tsSecondPartitionFine);
+        enqueueWrite(TABLE_CONSERVATIVE, largestFirstPartitionCoarse);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertLowestFinePartitionInSweepableTimestampsEquals(tsPartitionFine(TS));
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertLowestFinePartitionInSweepableTimestampsEquals(tsPartitionFine(TS));
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertLowestFinePartitionInSweepableTimestampsEquals(tsPartitionFine(TS));
+
+        // next sweep goes into a new coarse partition
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertNoEntriesInSweepableTimestampsBeforeImmutableTimestamp();
+    }
+
+    @Test
+    public void sweepableTimestampsGetsScrubbedWhenLastSweptProgressesInNewCoarsePartition2() {
```

Comment:
nit: `2`?

Reply:
Planned to rename this later and forgot 👎